app/scripts/services/user.js

```diff
@@ -84,7 +83,7 @@ angular.module('bballApp')
       if (!username) {
         defer.reject("Username is not valid");
       } else if (users.hasOwnProperty(username.toLowerCase())) {
-        currentUser = users[username.toLowerCase()];
+        this.currentUser = users[username.toLowerCase()];
```

Comment:
Should not define an object on 'this' (or any service) as you then can not control the assignment or modification of that object. e.g. If I want to set the user but validate that the user object has a username then I would create a function called setUser() so that all user assigned goes through this function. I can then place any validation code in this function.